### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by k-takata**
Please review carefully.

(Some trailing spaces were removed because of the setting of EditorConfig.)

### CODE REVIEW

File: parsers/cpreprocessor.h [hunk 1]
```diff
@@ -78,7 +78,7 @@ extern int cppGetc (void);
 extern int cppSkipOverCComment (void);
 
 /* notify the external parser state for the purpose of conditional
-   branche choice. The CXX parser stores the block level here. */
+   branch choice. The CXX parser stores the block level here. */
```
Comment (k-takata) [inline parsers/cpreprocessor.h]: Or should be `branches`?

File: parsers/cxx/cxx_parser.c [hunk 2]
```diff
@@ -973,7 +973,7 @@ static bool cxxParserParseClassStructOrUnionInternal(
 			(!(uInitialKeywordState & CXXParserKeywordStateSeenTypedef))
 		)
 	{
-		CXX_DEBUG_PRINT("Maybe KnR funciton definition?");
+		CXX_DEBUG_PRINT("Maybe KnR function definition?");
```
Comment (k-takata) [inline parsers/cxx/cxx_parser.c]: "KnR" or "K&R"?
It's inconsistent in this file.

File: parsers/cxx/cxx_token_chain.h [hunk 3]
```diff
@@ -190,7 +190,7 @@ vString * cxxTokenChainJoinRange(
 		unsigned int uFlags
 	);
 
-// Treat the tochek chain tc as a comma separated sequence
+// Treat the token chain tc as a comma separated sequence
```
Comment (k-takata) [inline parsers/cxx/cxx_token_chain.h]: Is this fix okay?

**Comment by k-takata** [comment]
Another questions.

https://github.com/universal-ctags/ctags/blob/dda51569a8d0bdf360ce5c15f643e113e6f836be/parsers/make.c#L299
What is this "endef"? Mistake of "endif"?

https://github.com/universal-ctags/ctags/blob/dda51569a8d0bdf360ce5c15f643e113e6f836be/parsers/ocaml.c#L418
Is "consecuting" a mistake of "consecutive"?

File: parsers/cpreprocessor.h (hunk 1 above)
Comment (masatake) [inline parsers/cpreprocessor.h]: I don't know. But `branch` may be o.k.

File: parsers/cxx/cxx_parser.c (hunk 2 above)
Comment (masatake) [inline parsers/cxx/cxx_parser.c]: Maybe `K&R` is better. However, the string KnR is used in a name of cunction.  I think both words are acceptable.

**Comment by masatake** [comment]
I have introduced many of wrong spellilng.

I must build infrastructure for running aspell on

1. comments
2. literal strings
3. commit log
4. subwords in identifiers in C source code
5. rst docs

The infrastructure can be integrated to travis.

**Comment by masatake** [comment]
`endef` is endef:-)

Quoted from the info document of GNU Make:
```
   define IMMEDIATE
       DEFERRED
     endef
```

File: parsers/cxx/cxx_parser.c (hunk 2 above)
Comment (k-takata) [inline parsers/cxx/cxx_parser.c]: I will leave this to @pragmaware.

**Comment by k-takata** [comment]
> endef is endef:-)

Oh, I didn't know that.

> Is "consecuting" a mistake of "consecutive"?

Fixed.

**Comment by k-takata** [comment]
> I must build infrastructure for running aspell on

It would be nice, but I suppose it will detect many false positives and need workarounds for them.
I think it's better to leave this for a task after the first release.

BTW, doesn't Emacs have a plugin for spell checking?
Vim has a built-in spell checker and I used it for this PR.

**Comment by masatake** [comment]
Emacs has a spell checker. However, I sometimes forget to run it before making a PR.
I find I want a unified interface for running a source code tree.

```
$ git checkout devel
$ git spell-check --ignore=Tmain,Units --dictionary=tags,extra.dict --commit-log --string-literal=C,Shell --comments=C,Shell --identifiers=C,Shell master...
commit log...o.k.
string literal...o.k.
comments...o.k.
identifiers...o.k.
...
```

So I can run git spell-check before making a pull request:.
I strongly belive someone on github already develop such software.

> I think it's better to leave this for a task after the first release.
You are right. However, `git spell-check` is very attractive idea for me.

**Comment by k-takata** [comment]
I forgot to check man/ctags.1.rst.in.
Updated.

BTW, are the command line options only described in man/ctags.1.rst.in and not in docs?

**Comment by masatake** [comment]
ctags.1.rst.in is not updated though we should do.
Could you review it if I update it?

**Comment by k-takata** [comment]
> Could you review it if I update it?

Yes.

Then, will an html file generated from ctags.1.rst.in be also available on http://docs.ctags.io/en/latest/ ?
Or, will another rst file about command line options be created in the docs directory?

FYI, I found an interesting project for detecting typo: https://github.com/chakki-works/typot .
A Japanese post is here: http://qiita.com/icoxfog417/items/5edb32b34dfb7886d426 .

**Comment by masatake** [comment]
> Yes.

Thank you. 
 I will update ctags.1.rst.in incrementally. Each time, I will add you "Reviewers".

>Then, will an html file generated from ctags.1.rst.in be also available on http://docs.ctags.io/en/latest/ ?
> Or, will another rst file about command line options be created in the docs directory?

```
$ make man/ctags.1
```
Generated the man page if rst2man is available.

The man page is not at http://docs.ctags.io.

> 
> FYI, I found an interesting project for detecting typo: https://github.com/chakki-works/typot .
> A Japanese post is here: http://qiita.com/icoxfog417/items/5edb32b34dfb7886d426 .

Very interesting. We should use it.
I also try my ideas for spell checking. This will not run on Github or CIs behind it. Mine is just run in local. 


Anyway I'm bit busy these days. So, please wait the initial review  request for awhile.

**Pull request merged**